runtime/doc/lua.txt

```diff
@@ -860,7 +860,7 @@ vim.spell.check({str})                                     *vim.spell.check()*
       • {str}  (string)
 
     Return: ~
-        `{[1]: string, [2]: string, [3]: string}[]` List of tuples with three
+        ({[1]: string, [2]: string, [3]: string}[]) List of tuples with three
```

Comment:
This is sort of a regression, however I think we should either surround all types with backticks or not at all.

@clason?

Reply:
I think the backticks help with reading complex return types, but I agree: since the vast majority of type annotations uses parens, deviating adds (a little) friction. So they're not worth keeping around, especially if this makes docgen easier/more robust.

Reply:
+1 for backticks. Parens are not highlighted.

If we use backticks for all types, some will lose links to helptags (e.g. |TSNode|), but I think having links would be useful for some types.

Reply:
The backticks are trivial to add, it's really a matter of preference. I would vote to add them since it reduces the change of vimdoc parse errors as types can be complicated expressions. The downside would that we wouldn't be able to use the types as tags.

Don't need to make a decision now, I've not included them in this PR to reduce the diff.

Reply:
Either is fine with me, I trust your opinion on which is more valuable. (The links can always be moved into the annotation comment, so we don't _necessarily_ lose them.)

Reply:
Ok, we can hold off for now and see how the vimdoc parser copes.

Reply:
I'm fine with adding backticks now, for the record.

Reply:
For links: it seems that `CTRL-]` on the following anchor can work, correctly jumps to `*TSNode*`:

```
|`TSNode`|

:help |`TSNode`|
```

However, this doesn't work (while `CTRL-]` can jump):
```
:help `|TSNode|`
```